compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -257,13 +257,12 @@ pub fn target_machine_factory(
     };
     let debuginfo_compression = SmallCStr::new(&debuginfo_compression);
 
-    let should_prefer_remapped_for_split_debuginfo_paths =
-        sess.should_prefer_remapped_for_split_debuginfo_paths();
+    let should_prefer_remapped_paths = sess.should_prefer_remapped_for_codegen();
 
     Arc::new(move |config: TargetMachineFactoryConfig| {
         let path_to_cstring_helper = |path: Option<PathBuf>| -> CString {
             let path = path.unwrap_or_default();
-            let path = if should_prefer_remapped_for_split_debuginfo_paths {
+            let path = if should_prefer_remapped_paths {
```

Comment:
I can't quite figure out if this path is emitted into something or is just the path where LLVM will actually put the file on disk. Or both 🙂

We'd only want to remap if it is written into debuginfo.

Reply:
I think to remember that it is emitted into debuginfo, or at least it was since this place already had the remapping in place before I updated it for the trim-path feature.

Reply:
Yep, you're right. This is the path that ends up in the `DW_AT_GNU_dwo_name` field of the compilation unit DIE.